File: app/src/main/java/com/coderred/andclaw/data/ChannelConfig.kt
package com.coderred.andclaw.data

data class ChannelConfig(
    val whatsappEnabled: Boolean = true,
    val telegramEnabled: Boolean = false,
    val telegramBotToken: String = "",
    val discordEnabled: Boolean = false,
    val discordBotToken: String = "",
    val discordGuildAllowlist: String = "",
    val discordRequireMention: Boolean = true,
)

enum class ChannelConnectionStatus {
    DISCONNECTED,
    CONNECTING,
    CONNECTED,
    ERROR,
}

data class ChannelStatus(
    val whatsapp: ChannelConnectionStatus = ChannelConnectionStatus.DISCONNECTED,
    val telegram: ChannelConnectionStatus = ChannelConnectionStatus.DISCONNECTED,
    val discord: ChannelConnectionStatus = ChannelConnectionStatus.DISCONNECTED,
)

File: windows/AndClaw.Windows/Models/ChannelConfig.cs
namespace AndClaw.Windows.Models;

public sealed class ChannelConfig
{
    public bool WhatsAppEnabled { get; set; } = true;
    public bool TelegramEnabled { get; set; }
    public bool DiscordEnabled { get; set; }
    public string TelegramBotToken { get; set; } = string.Empty;
    public string DiscordBotToken { get; set; } = string.Empty;
}
